site/js/index.mjs

```diff
@@ -201,28 +201,42 @@ function addQType(evt) {
 
 function showQueryError(ev) {
     const searchTermElement = document.querySelector("#searchterms");
-    
-    searchTermElement.classList.add("error");            
-            
+    const searchButtonElement = document.querySelector("#basic-addon2");
+
+    searchTermElement.classList.add("error");
+
+    // Mark the search term that caused the error with the error color.
+    for (const offendingTerm of ev.detail.offendingSearchTerms) {
+        const selectedOptions = document.querySelectorAll(`.optioncontainer[data-qvalue="${offendingTerm.value}"]`);
+        if (selectedOptions.length > 0) {
+            selectedOptions[0].classList.add("error");
+        }
+    }
```

Comment:
This is better written as: 

```
ev.detail.offendingSearchTerms.forEach((term) => 
    document.querySelector(`.optioncontainer[data-qvalue="${term.value}"]`)?.classList.add("error"));
```

Reply:
Besides, i am allergic against for, for-in and for-of loops. Better use functions whenever possible.

Reply:
I've changed it.

(I personally prefer for loops, since they make it easier to spot loops at a glance in code, i.e. when debugging something, that's why I used them. But of course, I'll try to use functions in your projects whenever possible)